src/components/check-your-phone/check-your-phone-controller.ts

```diff
@@ -115,6 +72,104 @@ export function checkYourPhonePost(
   };
 }
 
+async function handleMfaChange(
+  intent: Intent,
+  newPhoneNumber: string,
+  updateInput: UpdateInformationInput,
+  sessionDetails: any,
+  req: Request,
+  service: CheckYourPhoneServiceInterface
+): Promise<boolean> {
+  if (intent === INTENT_CHANGE_PHONE_NUMBER) {
+    return handleChangePhoneNumber(
+      newPhoneNumber,
+      updateInput,
+      sessionDetails,
+      req,
+      service
+    );
+  } else if (intent === INTENT_ADD_MFA_METHOD) {
+    return handleAddMfaMethod(
+      newPhoneNumber,
+      updateInput,
+      sessionDetails,
+      req,
+      service
+    );
+  } else {
+    // TODO: Confirm with UCD which error we should display
+    throw Error(`Unknown phone verification intent ${intent}`);
+  }
+}
+
+async function handleChangePhoneNumber(
+  newPhoneNumber: string,
+  updateInput: UpdateInformationInput,
+  sessionDetails: any,
+  req: Request,
+  service: CheckYourPhoneServiceInterface
+): Promise<boolean> {
+  const smsMFAMethod = req.session.mfaMethods.find(
+    (mfa) => mfa.method.mfaMethodType === "SMS"
+  );
+  if (smsMFAMethod) {
+    smsMFAMethod.method.endPoint = newPhoneNumber;
+    updateInput.mfaMethod = smsMFAMethod;
+    return await service.updatePhoneNumberWithMfaApi(
+      updateInput,
+      sessionDetails
+    );
+  } else {
+    // TODO: Confirm with UCD which error we should display
+    throw Error(`No existing MFA method for: ${req.session.user.email}`);
+  }
+}
+
+async function handleAddMfaMethod(
+  newPhoneNumber: string,
+  updateInput: UpdateInformationInput,
+  sessionDetails: any,
+  req: Request,
+  service: CheckYourPhoneServiceInterface
+): Promise<boolean> {
+  const smsMFAMethod = req.session.mfaMethods.find(
+    (mfa) => mfa.priorityIdentifier === "DEFAULT"
+  );
+  if (smsMFAMethod) {
+    smsMFAMethod.method.endPoint = newPhoneNumber;
+    updateInput.credential = "no-credentials";
+    updateInput.mfaMethod = {
+      ...smsMFAMethod,
+      mfaIdentifier: smsMFAMethod.mfaIdentifier + 1,
+      priorityIdentifier: "BACKUP",
+      method: {
+        mfaMethodType:
+          smsMFAMethod.method.mfaMethodType === "SMS" ? "AUTH_APP" : "SMS",
+        endPoint: newPhoneNumber,
+      },
+      methodVerified: true,
+    };
+    return await service.addMfaMethodService(updateInput, sessionDetails);
+  }
```

Comment:
Do you want to throw an error here too like in `handleChangePhoneNumber`?

Reply:
Yes. Done.